app/main.py

```diff
@@ -1,2 +1,55 @@
+from typing import Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.size = 0
+        self.threshold = int(self.capacity * 2 / 3)
+        self.hash_table = [None for _ in range(self.capacity)]
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        if self.size == self.threshold:
+            self.resize()
+        hash_ = hash(key)
+        index_ = hash_ % self.capacity
+        if self.hash_table[index_] is None:
+            self.hash_table[index_] = (key, hash_, value)
+            self.size += 1
+            return
+        if self.hash_table[index_][0] == key:
+            self.hash_table[index_] = (key, hash_, value)
+            return
+        for _ in range(self.capacity):
```

Comment:
It's safe to use `whlie True` here. By the way, you may combine the contents of this loop with the code above to avoid code duplication.